pmd-plsql/src/main/java/net/sourceforge/pmd/lang/plsql/ast/PLSQLNode.java

```diff
@@ -4,27 +4,43 @@
 
 package net.sourceforge.pmd.lang.plsql.ast;
 
-import net.sourceforge.pmd.lang.ast.Node;
-import net.sourceforge.pmd.lang.ast.NodeStream;
+import net.sourceforge.pmd.lang.ast.impl.javacc.JjtreeNode;
 import net.sourceforge.pmd.lang.symboltable.Scope;
 import net.sourceforge.pmd.lang.symboltable.ScopedNode;
 
-public interface PLSQLNode extends Node, ScopedNode {
+public interface PLSQLNode extends ScopedNode, JjtreeNode<PLSQLNode> {
 
     /** Accept the visitor. **/
     Object jjtAccept(PLSQLParserVisitor visitor, Object data);
 
     @Override
     Scope getScope();
 
-    void setScope(Scope scope);
-
-    @Override
-    PLSQLNode getChild(int index);
-
-    @Override
-    PLSQLNode getParent();
-
-    @Override
-    NodeStream<? extends PLSQLNode> children();
+    /**
+     * Return node image converted to the normal Oracle form.
+     *
+     * <p>
+     * Normally this is uppercase, unless the names is quoted ("name").
+     * </p>
+     */
+    default String getCanonicalImage() {
```

Comment:
hm...yes, these methods are not ideal, too generic like "getImage". but this time on plsql...